### ISSUE TITLE
Can I pay you for help making ClickOnce publish via Command Line?

### ISSUE THREAD

**Issue by TonyValenti** [sayedihashimi/publish-samples#5]
If so, please email TV@MatterMovers.net .  I could really use the help.

**Comment by sayedihashimi** [comment]
@TonyValenti in Visual Studio the team has done some work to simplify the Click Once publish experience. With the latest version of VS2019, when you publish with ClickOnce a publish profile is created (.pubxml file under the project Properties folder). Once that profile is created you should be able to automate publish with a command similar to:

`msbuild.exe [path to project file] /p:DeployOnBuild=true /p:PublishProfile=[name-of-publish-profile]`

More info on publish profiles can be found at https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/visual-studio-publish-profiles?view=aspnetcore-5.0.

Let me know if that helps you or not.

**Comment by TonyValenti** [comment]
Hi @sayedihashimi !
Thanks so much.  I was able to figure it out.  Here's what the problem was:

1.  Before running MSBuild, I need to do a dotnet restore with the following command:
```dotnet restore -r win-x64 $Solution```
When doing the restore, I must have VS closed otherwise, VS will essentially do an "un-restore" why the restore is happening.

2.  Next, I needed to do MSBuild with the following command:
```& $MsBuild "$Project" /t:publish /p:DeployOnBuild=true "/p:PublishProfile=$PublishingProfile" /p:PublishDir="$OutputDir" /property:Configuration=Release```
